tests/test_unit/test_navigation.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import pytest
+import xarray as xr
+
+from movement.analysis import navigation
+
+
+@pytest.fixture
+def mock_dataset():
+    """Return a mock DataArray containing four known head orientations."""
+    time = np.array([0, 1, 2, 3])
+    individuals = np.array(["individual_0"])
+    keypoints = np.array(["left_ear", "right_ear", "nose"])
+    space = np.array(["x", "y"])
+
+    ds = xr.DataArray(
+        [
+            [[[1, 0], [-1, 0], [0, -1]]],  # time 0
+            [[[0, 1], [0, -1], [1, 0]]],  # time 1
+            [[[-1, 0], [1, 0], [0, 1]]],  # time 2
+            [[[0, -1], [0, 1], [-1, 0]]],  # time 3
+        ],
+        dims=["time", "individuals", "keypoints", "space"],
+        coords={
+            "time": time,
+            "individuals": individuals,
+            "keypoints": keypoints,
+            "space": space,
+        },
+    )
+    return ds
+
+
+def test_compute_head_direction_vector(mock_dataset):
+    """Test that the correct head direction vectors
+    are computed from a basic mock dataset.
+    """
+    # Test that validators work
+    with pytest.raises(TypeError):  # Catch incorrect datatype
+        np_array = [
+            [[[1, 0], [-1, 0], [0, -1]]],
+            [[[0, 1], [0, -1], [1, 0]]],
+            [[[-1, 0], [1, 0], [0, 1]]],
+            [[[0, -1], [0, 1], [-1, 0]]],
+        ]
+        navigation.compute_head_direction_vector(
+            np_array, "left_ear", "right_ear"
+        )
+
+    with pytest.raises(AttributeError):  # Catch incorrect dimensions
+        mock_dataset_keypoint = mock_dataset.sel(keypoints="nose", drop=True)
+        navigation.compute_head_direction_vector(
+            mock_dataset_keypoint, "left_ear", "right_ear"
+        )
+
+    with pytest.raises(ValueError):  # Catch identical left and right keypoints
+        navigation.compute_head_direction_vector(
+            mock_dataset, "left_ear", "left_ear"
+        )
+
+    with pytest.raises(ValueError):  # Catch incorrect spatial dimensions
+        time = np.array([0])
+        individuals = np.array(["individual_0"])
+        keypoints = np.array(["left", "right"])
+        space = np.array(["x", "y", "z"])
+
+        ds = xr.DataArray(
+            [
+                [[[1, 0, 0], [-1, 0, 0]]],
+            ],
+            dims=["time", "individuals", "keypoints", "space"],
+            coords={
+                "time": time,
+                "individuals": individuals,
+                "keypoints": keypoints,
+                "space": space,
+            },
+        )
+
+        navigation.compute_head_direction_vector(ds, "left", "right")
+
+    # Test that output contains correct datatype, dimensions, and values
+    head_vector = navigation.compute_head_direction_vector(
+        mock_dataset, "left_ear", "right_ear"
+    )
+    known_vectors = np.array([[[0, 2]], [[-2, 0]], [[0, -2]], [[2, 0]]])
```

Comment:
To me these results make sense in our coordinate system. I've drawn the calculation by hand for the first timepoint (`left_ear=[1,0]`, `right_ear=[-1,0]`). The resulting head direction vector points down (in increasing y direction), so the `[0. 2]` result makes sense to me.
![Head-Direction-Vector](https://github.com/user-attachments/assets/f2dc5ed7-36b9-4364-893b-caf29e285d6e)